backend/src/main/scala/bloop/BloopClassFileManager.scala

```diff
@@ -216,19 +216,36 @@ final class BloopClassFileManager(
         ) => {
           clientTracer.traceTaskVerbose("copy new products to external classes dir") { _ =>
             val config = ParallelOps.CopyConfiguration(5, CopyMode.ReplaceExisting, Set.empty)
-            ParallelOps
-              .copyDirectories(config)(
-                newClassesDir,
-                clientExternalClassesDir.underlying,
-                inputs.ioScheduler,
-                enableCancellation = false,
-                inputs.logger
-              )
-              .map { walked =>
-                readOnlyCopyDenylist.++=(walked.target)
+            val clientExternalBestEffortDir =
+              clientExternalClassesDir.underlying.resolve("META-INF/best-effort")
+
+            // Deletes all previous best-effort artifacts to get rid of all of the outdated ones.
```

Comment:
Not at the moment unfortunately. Maybe I should add an option in metals to be able disable best-effort? It should be able to start the compilation with the files from last successful compilation, but I do not remember if that's the case to be honest (it should be)

Reply:
Do you mean that a new betasty directory is always spawned? It's fine if we remove them later on.

Reply:
It is always spawned, but I was talking about the new internal bloop directory being created (and the previous one not being deleted) when following failing compilation with a successful one. That's not a big issue (outside of spamming a little bit with files in .bloop), but I also saw that we delete the previous successful artifacts (.class and .tasty) altogether in client directory when ending up with betasty, which I think is what I wanted to write before, since you were asking about deleting old artifacts - sorry about that.

Reply:
Basically:
- we have a successful compilation, we end up with META-INF/betasty, semanticdb and regular classifies and tasty in the client directory (which metals uses)
- we follow this with an unsuccessful compilation with some kind of error, we end up with just META-INF/betasty, semanticdb (other files get deleted, I already found where that happens)

Reply:
The flow is a bit complex with the different directories, I was thinking of simplifying it, but didn't get around to it yet. You mean to fix it in the current PR? Let me know if you need any help.

Reply:
Yes, fixing it right now